core/server/apps/amp/tests/amp_content_spec.js

```diff
@@ -0,0 +1,124 @@
+    });
+
+    it('can render content from cache', function (done) {
+        var testData = {
+                html: 'Hello World',
+                updated_at: 'Wed Jul 27 2016 18:17:22 GMT+0200 (CEST)',
+                id: 1
+            },
+            ampCachedResult,
+            ampResult = ampContentHelper.call(testData),
+            amperizeCache = ampContentHelper.__get__('amperizeCache');
+
+        ampResult.then(function (rendered) {
+            should.exist(rendered);
+            should.exist(amperizeCache);
+            rendered.string.should.equal(testData.html);
+            amperizeCache.should.have.property('1469636242000_1');
+            // call it again, to make it fetch from cache
+            ampCachedResult = ampContentHelper.call(testData);
+            ampCachedResult.then(function (rendered) {
+                should.exist(rendered);
+                should.exist(amperizeCache);
+                amperizeCache.should.have.property('1469636242000_1');
+                rendered.string.should.equal(testData.html);
+                done();
+            });
+        }).catch(done);
+    });
+
+    it('fetches new AMP HTML if post was changed', function (done) {
+        var testData1 = {
+                html: 'Hello World',
+                updated_at: 'Wed Jul 27 2016 18:17:22 GMT+0200 (CEST)',
+                id: 1
+            },
+            testData2 = {
+                    html: 'Hello Ghost',
+                    updated_at: 'Wed Jul 30 2016 18:17:22 GMT+0200 (CEST)',
+                    id: 1
+                },
+            ampResult = ampContentHelper.call(testData1),
+            amperizeCache = ampContentHelper.__get__('amperizeCache');
+
+        amperizeStub.returns({parse: function () { }});
+        ampContentHelper.__set__('Amperize.parse', amperizeStub);
+
+        ampResult.then(function (rendered) {
+            should.exist(rendered);
+            should.exist(amperizeCache);
+            // amperizeStub.calledOnce.should.be.true();
+            amperizeCache.should.have.property('1469636242000_1');
+            rendered.string.should.equal(testData1.html);
+
+            // call it again with different values to fetch from Amperize and not from cache
+            ampResult = ampContentHelper.call(testData2);
+            ampResult.then(function (rendered) {
+                should.exist(rendered);
+                should.exist(amperizeCache);
+
+                // it still has the old value in cache
+                amperizeCache.should.not.have.property('1469636242000_1');
+                // but also the new one
+                amperizeCache.should.have.property('1469895442000_1');
+                rendered.string.should.equal(testData2.html);
+                done();
+            });
+        }).catch(done);
+    });
+
+    // TODO: stub Amperize to make this fast!
+    it.skip('can transforms img and iframe tags and gifs to amp-img, amp-iframe and amp-anim', function (done) {
```

Comment:
Yep, it's on purpose. I first want to stub Amperize, because this little assertion here is a performance killer. But tbh, I wasn't sure at all if I need to test it, because the transform tests are already done in Amperize itself. What do you think, @kirrg001?